### ISSUE TITLE
Bring in improvements from other gdnative libraries

### ISSUE THREAD

**Issue by rosshadden** [pragmagic/godot-nim#66]
Some other gdnative libraries have raised the bar quite a bit in terms of usability / user experience since this project was first created. I will list some examples:

- [ECMAScript](https://github.com/GodotExplorer/ECMAScript), [C#](https://github.com/godotengine/godot/tree/master/modules/mono), and [Python](https://github.com/touilleMan/godot-python) add entries to the attach script dialog:
![2020-09-01-10-58-53](https://user-images.githubusercontent.com/669012/91868198-7d282100-ec42-11ea-9f8c-1ff4995374bd.png)
- None of the above require creating `.gdns` files (or embedding in scenes)
- Of these, only C# seems to require a manifest file with each included script, but it generates the file automatically so I only noticed because of the git diff. Based on what I saw when experimenting, I'm guessing the "create script" action calls a method in the C# library which it uses as a trigger to generate or add to the file.
- godot-python is self-contained and available in the asset store (I know this one is a stretch and don't think it could be done with godot-nim, at least not easily)
- C# and ECMAScript don't even require a `.gdnlib` file

Are we able to add any of these improvements to this project?

**Comment by sheepandshepherd** [comment]
None of those are NativeScript bindings, they're engine modules (except Python) that add fully-featured subclasses of `ScriptLanguage`/`Script`/`ScriptInstance` for their language. NativeScript is a different, much more minimalistic, thing. It doesn't allow adding languages to the Attach Script dialog or the internal editor, and strictly requires .gdnlib/.gdns files, because it's designed to never even know about your source files.

There are partial solutions for NativeScript bindings, like editor plugins or build scripts that auto-generate .gdns/.gdnlib files, or import source files *as* .gdns resources (for [C++](https://gitlab.com/turtlewit/godot_cpp_helper_plugin), [Rust](https://github.com/godot-rust/project-utils), and [D](https://github.com/sheepandshepherd/godot-d-plugin)).

I'm guessing the binding authors are unlikely to migrate to an engine module or PluginScript - it'd require a lot of rewriting and extra work, and would introduce some disadvantages over NativeScript. Improving NativeScript itself might be a more realistic option. Related issue: godotengine/godot-proposals#119

**Comment by ShalokShalom** [comment]
> None of those are NativeScript bindings, they're engine modules (except Python) that add fully-featured subclasses of `ScriptLanguage`/`Script`/`ScriptInstance` for their language. 

Wait, also EcmaScript?

**Comment by sheepandshepherd** [comment]
Yes, that one is an engine module and doesn't use GDNative or NativeScript.
